### PR TITLE
CONJ-412: tinyInt1isBit is not applied in method columnTypeClause

### PR DESCRIPTION
**Pull request by mikecelis**
The method columnTypeClause has been modified to use the tinyInt1isBit flag. This makes sure things such as hibernate validation don't fail. It's solved by checking if the tinyInt1isBit flag is true and when it is true it will add an if to the sql each time the columnName is used to replace any occurences of 'tinyint(1)' with 'BIT'. If the flag is false the columnName will be added unmodified. This has been tested with an application using hibernate-validator-5.2.4.Final and MariaDB 10.1.2

### CODE REVIEW

File: src/main/java/org/mariadb/jdbc/MariaDbDatabaseMetaData.java [hunk 1]
```diff
@@ -86,12 +86,16 @@ public MariaDbDatabaseMetaData(Connection connection, String user, String url) {
     }
 
     /* Remove length from column type spec,convert to uppercase,  e.g  bigint(10) unsigned becomes BIGINT UNSIGNED */
-    static String columnTypeClause(String columnName) {
+    String columnTypeClause(String columnName) {
+        String parsedColumnName = parseColumnName(columnName);
+        return " UCASE(IF( " + parsedColumnName + " LIKE '%(%)%', CONCAT(SUBSTRING( " + parsedColumnName + ",1, LOCATE('(',"
+                + parsedColumnName + ") - 1 ), SUBSTRING(" + parsedColumnName + ",1+locate(')'," + parsedColumnName + "))), "
+                + parsedColumnName + "))";
+    }
```
Comment (rusher) [inline src/main/java/org/mariadb/jdbc/MariaDbDatabaseMetaData.java]: That's correct, but make request very big.
Would be better just to add just another IF, like : 

    static String columnTypeClause(int dataTypeMappingFlags, String columnName) {
        if ((dataTypeMappingFlags  & MariaSelectResultSet.TINYINT1_IS_BIT) > 0) {
            return " UCASE( "
                    + " IF(" + columnName + " = 'tinyint(1)', 'BIT', "
                    + " IF( " + columnName + " LIKE '%(%)%', CONCAT(SUBSTRING( " + columnName + ",1, LOCATE('(',"
                    + columnName + ") - 1 ), SUBSTRING(" + columnName + ",1+locate(')'," + columnName + "))), "
                    + columnName + ")))";
        }
        return " UCASE(IF( " + columnName + " LIKE '%(%)%', CONCAT(SUBSTRING( " + columnName + ",1, LOCATE('(',"
                + columnName + ") - 1 ), SUBSTRING(" + columnName + ",1+locate(')'," + columnName + "))), "
                + columnName + "))";
    }

Comment (mikecelis) [inline src/main/java/org/mariadb/jdbc/MariaDbDatabaseMetaData.java]: Okay, I've modified and pushed the code.

**Comment by rusher** [comment]
Hi @mikecelis,

Thanks for you contribution.

Similar to other open source projects, the MariaDB Foundation needs to have shared ownership of all code that is included in the MariaDB distribution. The easiest way to achieve this is by submitting your code under the BSD-new license. (The other alternative is to sign the code contribution agreement which can be found here: https://mariadb.com/kb/en/mariadb/mca/)

Please indicate in a comment below that you are contributing your new code of the whole pull request under the BSD-new license or that you have filled out the contribution agreement and sent it.

Thanks,
diego

**Comment by mikecelis** [comment]
We're contributing the code of this whole pull request under the BSD-new license.

**Pull request merged**